### DISCUSSION TITLE
Trying to add Text to globe

### DISCUSSION THREAD

**Discussion by miles-tbe** [pelicanmapping/osgearth#1926] [Support Q&A] [answered]
I have this code here:
```cpp 
const std::string text = Stringify () << "Location: " << lat << ", " << lon; 

osgText::Text *label = new osgText::Text ();
label->setText (text);
label->setColor (Color::Red);
label->setPosition (osg::Vec3 (X, Y, Z));
label->setAlignment (label->CENTER_CENTER);
label->setAutoRotateToScreen (true);
label->setCharacterSizeMode (label->SCREEN_COORDS);
label->setCaracterSize (20.0f);
label->setFont ("arialbd.ttf");
Group->addChild (label);
```
I grabbed this code from the osgearth_atlas.cpp example. The Group I have has been added to the MapHelperNode variable. 

What I am seeing is the text only displaying when I am tilted towards the sky. Whenever I am facing the globe itself, it cannot be seen. If I hit 'w' for the wireframe I can see that is still being drawn in this same location, but it is being obstructed in some way and I am not sure why.

**Comment by miles-tbe** [comment, marked as answer]
I figured this out on my own. I needed to use osgText::Text3D.

**Comment by chenguanga** [comment]
Why do not use osgEarth::LabelNode?
